### PR TITLE
Feat: add identity provider to DTR

### PR DESCRIPTION
**Pull request by tom-rm-meyer-ISST**
## Description

Add identity provider configuration for the PURIS backend:
- EDC client with only read access that is added to the data asset's address
- PURIS client with all permissions for shell descriptor updates etc.

Changes:
- Add OAuth2ClientInterceptor with DTR configuration that is conditionally binded
- DataAddress in Asset registration for DTR conditionally contains oauth2 keys
- Test for OAuth2ClientInterceptor
- Helm chart
  - add dtr idp config
  - add missing frameworkagreement.use in values
- local setup: 
  - centralize keycloak with two additional realms to configure the idp per partner
  - keycloak setup for miw does not need anymore a jq injection because it now uses environment variables
  - separate keycloak and miw properties

Note: changes without formatting `git diff --stat -w main` results in delta of 6062 lines of code (loc) of which the following are no code changes:
- 5010 loc: keycloak realms for customer and supplier as json
- 41 loc: plain markdown documentation
- 60 loc: plain license headers
result (not even considering inline documentation like javadoc): **951 loc**

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: local/cleanup.sh [hunk 1]
```diff
@@ -22,6 +22,7 @@
 
 docker compose down -v
 docker compose -f docker-compose-infrastructure.yaml down -v
+docker compose -f docker-compose-idp.yaml down -v
```
Comment (eschrewe) [inline local/cleanup.sh:25]: Does this file exist in this project?

**Review by eschrewe — CHANGES_REQUESTED** [review]
Very well done, 
just some minor remarks from my side.

File: local/docker-compose.yaml [hunk 2]
```diff
@@ -238,8 +238,13 @@ services:
...
       SPRING_DATASOURCE_USERNAME: ${PG_USER}
       SPRING_DATASOURCE_PASSWORD: ${PG_PW}
-      SPRING_PROFILES_ACTIVE: local
+      REGISTRY_IDM_OWNING_TENANT_ID: ${SUPPLIER_BPNL}
+      REGISTRY_USE_GRANULAR_ACCESS_CONTROL: false
       JAVA_TOOL_OPTIONS: "-Xms512m -Xmx1024m"
+      # outcomment SPRING_PROFILES_ACTIVE if idp should be used
+      SPRING_PROFILES_ACTIVE: local
+      REGISTRY_IDM_PUBLIC_CLIENT_ID: ${KC_READ_CLIENT_ID}
+      SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_ISSUER_URI: "http://keycloak-supplier:10080/realms/Catena-X"
```
[the remark this answers is not in the document]
Comment (tom-rm-meyer-ISST) [reply local/docker-compose.yaml]: Fixed. 

Local Deployment has been forgotten to be updated at that point:
- enable security usage by outcommenting `SPRING_PROFILES_ACTIVE`
- correct values for `REGISTRY_IDM_PUBLIC_CLIENT_ID` and `SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_ISSUER_URI`

customer container missed the config overall. Thus added.

File: local/cleanup.sh (hunk 1 above)
Comment (tom-rm-meyer-ISST) [reply local/cleanup.sh:25]: Great catch. Missed to remove that. Used initially a separate docker compose file to easier spin up the idp only during testing and configuration of the realm.

Removed / Fixed :)

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/ddtr/logic/DtrAdapterService.java
[the remark this answers is not in the document]
Comment (tom-rm-meyer-ISST) [reply backend/src/main/java/org/eclipse/tractusx/puris/backend/common/ddtr/logic/DtrAdapterService.java:1]: merged main

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/security/DtrSecurityConfiguration.java [hunk 3]
```diff
@@ -44,17 +44,17 @@ public class DtrSecurityConfiguration {
...
+    private String purisClientId;
     /**
      * client secret of the puris client with write access for DTR
      **/
     @Value("${puris.dtr.idp.puris-client.secret}")
-    String purisClientSecret;
+    private String purisClientSecret;
     /**
      * grant_type. Currently only client_credentials is supported
      **/
```
Comment (eschrewe) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/common/security/DtrSecurityConfiguration.java:60]: Grant type still not private

**Review by eschrewe — CHANGES_REQUESTED** [review]
Almost there :)

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/common/security/DtrSecurityConfiguration.java (hunk 3 above)
Comment (tom-rm-meyer-ISST) [reply backend/src/main/java/org/eclipse/tractusx/puris/backend/common/security/DtrSecurityConfiguration.java:60]: ah missed that one. Thanks, resolved!

**Review by eschrewe — APPROVED** [review]
LGTM, thank you very much :)

**Review by mhellmeier — APPROVED** [review]
As described by @tom-rm-meyer-ISST in the description, the changes in this PR based on LOC are minimal. This is due to the autogenerated realm files like [this](https://github.com/eclipse-tractusx/puris/blob/d8b53ac1f14b105f59c31d3b0c011d1583b854cf/local/keycloak/customer/Customer-realm.json) one or [this](https://github.com/eclipse-tractusx/puris/blob/d8b53ac1f14b105f59c31d3b0c011d1583b854cf/local/keycloak/supplier/Supplier-realm.json) one. Therefore, the actual LOC are under 1000.

**Pull request merged**